sonic_platform_base/device_base.py

```diff
@@ -0,0 +1,66 @@
+#
+# device_base.py
+#
+# Abstract base class for interfacing with a generic type of platform
+# peripheral device in SONiC
+#
+
+class DeviceBase(object):
+    """
+    Abstract base class for interfacing with a generic type of platform
+    peripheral device
+    """
+
+    def get_presence(self):
+        """
+        Retrieves the presence of the device
+
+        Returns:
+            bool: True if device is present, False if not
+        """
+        raise NotImplementedError
+
+    def get_model(self):
+        """
+        Retrieves the model number (or part number) of the device
+
+        Returns:
+            string: Model/part number of device
+        """
+        raise NotImplementedError
+
+    def get_serial(self):
+        """
+        Retrieves the serial number of the device
+
+        Returns:
+            string: Serial number of device
+        """
+        raise NotImplementedError
+
+    def get_status(self):
+        """
+        Retrieves the operational status of the device
+
+        Returns:
+            A boolean value, True if device is operating properly, False if not
+        """
+        raise NotImplementedError
+
+    def get_change_event(self, timeout=0):
+        """
+        Returns a dictionary containing all devices which have experienced a
+        change
+
+        Args:
+            timeout: Timeout in milliseconds (optional). If timeout == 0,
+                this method will block until a change is detected.
+
+        Returns:
+            (bool, dict):
+                - True if call successful, False if not;
+                - Dict where key is device ID and value is device event,
+                  status='1' represents device inserted,
+                  status='0' represents device removed. Ex. {'0': '1', '1': '0'}
+        """
+        raise NotImplementedError
```

Comment:
Just wanted to get confirmation that if a platform takes a Baseboard Management Controller (BMC) based approach (to handle thermals, events etc), these APIs can remain unimplemented..

Reply:
The intention here is for daemons to receive notifications of changes in platform hardware state in order to update SONiC's view of device state. For instance, if a tranceiver, PSU or fan tray is added or removed. Can you provide an example of an event handled by the BMC which SONiC would not need to be made aware of?